.github/workflows/rust.yml

```diff
@@ -73,17 +73,17 @@ jobs:
       run: echo RUSTFLAGS='-Dwarnings -Dclippy::all -Dclippy::pedantic' >> $GITHUB_ENV
     - run: rustup update
     - run: cargo version
-    - run: rustup target add ${{ matrix.target }}
+    - run: rustup target add ${{ matrix.sys.target }}
     - run: rustup target add wasm32-unknown-unknown
-    - if: matrix.target == 'aarch64-unknown-linux-gnu'
-      run: sudo apt-get update && sudo apt-get -y install gcc-aarch64-linux-gnu g++-aarch64-linux-gnu
-    - run: cargo clippy --all-targets --target ${{ matrix.target }}
+    - if: matrix.sys.os == 'ubuntu-latest-16-cores'
+      run: sudo apt-get update && sudo apt-get -y install gcc-aarch64-linux-gnu g++-aarch64-linux-gnu libudev-dev
+    - run: cargo clippy --all-targets --target ${{ matrix.sys.target }}
     - run: make build-test
-    - if: startsWith(matrix.target, 'x86_64')
+    - if: startsWith(matrix.sys.target, 'x86_64')
```

Comment:
This guard is wrong really. macos-latest is arm64 and we can run tests on it. We just can't run tests on arm64 on linux, without emulation and we don't have emulation setup. We have access to arm64 linux runners too so we could run tests if we wanted to now. Mentioning for a follow up PR.

Reply:
> We just can't run tests on arm64 on linux, without emulation

Could you elaborate more on this please?